library/src/webapp/js/portal/portal.sidebar.sites.js

```diff
@@ -5,16 +5,18 @@ class SitesSidebar {
     this._i18n = config?.i18n;
     this._element = element;
     this._lessonsSubpageData = config?.lessonsSubpageData;
+    this._pinnedSiteList = document.getElementById("pinned-site-list");
+    this._recentSiteList = document.getElementById("recent-site-list");
+    this._currentSite = config.currentSite;
 
     const sitesListItems = element.querySelectorAll(".site-list-item");
-    //sitesListItems.forEach(sitesListItem => new LessonsSubPageNavigation(this._lessonsSubpageData))
 
     const pinButtonElements = element.querySelectorAll(".site-opt-pin");
-    pinButtonElements.forEach((buttonEl) => new PinButton(buttonEl, { i18n: this._i18n?.pinButtons}));
+    pinButtonElements.forEach(buttonEl => new PinButton(buttonEl, { i18n: this._i18n?.pinButtons}));
 
     element.querySelectorAll(".site-description-button").forEach(buttonEl => new bootstrap.Popover(buttonEl));
 
-    document.addEventListener("site-pin-change", this.handlePinChange);
+    document.addEventListener("site-pin-change", this.handlePinChange.bind(this));
```

Comment:
Not sure in this case, but I prefer this.
```suggestion
    document.addEventListener("site-pin-change", (e) => this.handlePinChange(e));
```

Reply:
You can't just do that. You have to bind this to the class to be able to use the properties like _pinnedSiteList inside handlePinChange. We could just use document.getElementById at all times. This design was an attempt to be object oriented, so I understand the approach.